### PR TITLE
Use Junit Error Trace as Unit Result details

### PR DESCRIPTION
**Pull request by liuhaotian**
Harbormaster accept `details` for Unit Results. It would be more
informative for test report to have stack trace when test is failing.

For pass tests, the stack trace is null, which will be omitted by
`JSONObject`.

### CODE REVIEW

**Comment by sectioneight** [comment]
This looks good, thanks for contributing!

Could you please add a test using a JUnit XML fixture that has a stacktrace in it, and make sure it gets sent to the mock harbormaster correctly? I like to have these "integration" style tests to make sure that features aren't broken during refactors.

**Comment by sectioneight** [comment]
Awesome. One last step. As this appears to be your first contribution to Uber OSS, we need you to sign our SLA: http://t.uber.com/cla

File: src/test/java/com/uber/jenkins/phabricator/BuildIntegrationTest.java [hunk 1]
```diff
@@ -122,6 +134,15 @@ protected void assertLogContains(String needle, AbstractBuild build) throws IOEx
         assertThat(log.toString(), containsString(needle));
     }
 
+    protected void assertConduitRequest(JSONObject expectedRequestParams, String actualRequestBody) throws IOException, ConduitAPIException {
+        String conduitTestServerAddress = TestUtils.getTestServerAddress(conduit.getServer());
+        ConduitAPIClient conduitTestClient = new ConduitAPIClient(conduitTestServerAddress, TestUtils.TEST_CONDUIT_TOKEN);
+        HttpEntityEnclosingRequest request = (HttpEntityEnclosingRequest) conduitTestClient.createRequest("", expectedRequestParams);
+        String expectedRequestBody = EntityUtils.toString(request.getEntity());
+        assertEquals(expectedRequestBody, actualRequestBody);
+
```
Comment (sectioneight) [inline src/test/java/com/uber/jenkins/phabricator/BuildIntegrationTest.java]: Can remove this black line

File: src/test/java/com/uber/jenkins/phabricator/BuildIntegrationTest.java [hunk 2]
```diff
@@ -122,6 +134,15 @@ protected void assertLogContains(String needle, AbstractBuild build) throws IOEx
         assertThat(log.toString(), containsString(needle));
     }
 
+    protected void assertConduitRequest(JSONObject expectedRequestParams, String actualRequestBody) throws IOException, ConduitAPIException {
+        String conduitTestServerAddress = TestUtils.getTestServerAddress(conduit.getServer());
+        ConduitAPIClient conduitTestClient = new ConduitAPIClient(conduitTestServerAddress, TestUtils.TEST_CONDUIT_TOKEN);
+        HttpEntityEnclosingRequest request = (HttpEntityEnclosingRequest) conduitTestClient.createRequest("", expectedRequestParams);
+        String expectedRequestBody = EntityUtils.toString(request.getEntity());
```
Comment (sectioneight) [inline src/test/java/com/uber/jenkins/phabricator/BuildIntegrationTest.java]: Prefer newlines between test run and assertions

**Comment by sectioneight** [comment]
Very nicely done. Will merge after CLA

**Comment by liuhaotian** [comment]
✅ Removed unnecessary black line
✅ Added new line between test run and assertions
✅ Signed the CLA

Thanks for reviewing my pull request, Aiden!

**Comment by sectioneight** [comment]
👍 will merge after travis

**Pull request merged**